autonomi/src/networking/driver/task_handler.rs

```diff
@@ -489,6 +525,14 @@ impl TaskHandler {
             responder
                 .send(Ok(None))
                 .map_err(|_| TaskHandlerError::NetworkClientDropped(format!("{id:?}")))?;
+        // Get Merkle candidate quote case
+        } else if let Some(responder) = self.get_merkle_candidate_quote.remove(&id) {
```

Comment:
this now becomes a longer and longer if-else clause now.
ideally, a general req/resp dispatcher shall be used instead